src/hkdf/hkdf.ts

```diff
@@ -0,0 +1,40 @@
+import { ProviderCrypto } from "../provider";
+import { KeyUsages } from "../types";
+import { BufferSourceConverter } from "../utils";
+
+export abstract class HkdfProvider extends ProviderCrypto {
+
+  public name = "HKDF";
+  public hashAlgorithms = ["SHA-1", "SHA-256", "SHA-384", "SHA-512"];
+  public usages: KeyUsages = ["deriveKey", "deriveBits"];
+
+  public checkAlgorithmParams(algorithm: HkdfParams) {
+    // hash
+    this.checkRequiredProperty(algorithm, "hash");
+    this.checkHashAlgorithm(algorithm.hash as Algorithm, this.hashAlgorithms);
+
+    // salt
+    this.checkRequiredProperty(algorithm, "salt");
```

Comment:
'salt' is optional for HKDF

Reply:
Please check W3 spec
https://www.w3.org/TR/WebCryptoAPI/#dfn-HkdfParams

Reply:
Chrome throws error if `salt` or `info` params are missed
![image](https://user-images.githubusercontent.com/7333669/53679947-f3159a00-3ce4-11e9-98bc-d07aec244730.png)

Reply:
That is an odd choice by the W3C... The HKDF RFC says that both salt and info are optional and provides way to generate default values.

Sorry for the false comment, I wonder why the W3C chose not to follow the RFC.